### PR TITLE
Add telemetry listener and configs

### PR DESCRIPTION
**Pull request by rasika**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

**Comment by rasika** [comment]
Related PR https://github.com/ballerina-platform/ballerina-lang/pull/29118

File: package.json [hunk 1]
```diff
@@ -106,6 +106,11 @@
                     "type": "boolean",
                     "default": true,
                     "description": "Enable watching file change events of the project."
+                },
+                "ballerina.enableLanguageServerCrashReporter": {
+                    "type": "boolean",
+                    "default": true,
+                    "markdownDescription": "Enable crash reports to be sent to a Microsoft online telemtry service. Read privacy statement [here](https://privacy.microsoft.com/en-us/privacystatement)."
```
Comment (prabushi) [inline package.json:113]: I think we can't refer to Microsoft licensing policies as extensions come under their own licenses https://code.visualstudio.com/license
Shall we check on this?

Comment (rasika) [reply package.json:113]: Okay, I can remove that part. And wondering whether we can have a common config to enable/disable telemtry for ballerina plugin ?

Comment (prabushi) [reply package.json:113]: +1 for a common setting for the extension.

**Review by prabushi — APPROVED** [review]

**Pull request merged**